banno-enrollment-po/BANNO.ENROLLMENT.V1.POW

```diff
@@ -0,0 +1,772 @@
+ NEWPREFLOC                  = 0
+ REALLYDELETE                = ""
+ NEWACCESSTYPE               = 0
+ NEWACCESSACCT               = ""
+ NEWACCESSIDTYPE             = 0
+ NEWACCESSID                 = ""
+ NEWACCESSWD                 = 0
+ NEWACCESSDEP                = 0
+ NEWACCESSINQ                = 0
+ NEWACCESSFM                 = 0
+ ACCESSDESC                  = ""
+END [PROCEDURE]
+
+PROCEDURE SELECTPREF
+[* Dialog that prompts user to choose banno username if more 
+** than one HB preference is present.
+*]
+ IF PREFCOUNT=1 THEN
+  SELECTEDPREF=1
+ ELSE IF PREFCOUNT>1 THEN
+  DO
+   DIALOGSTART("Select Banno User",200%,0)
+   DIALOGPROMPTLISTSTART("Choose Banno User",1)
+   FOR PREFINDEX=1 TO PREFCOUNT
+    DO
+     DIALOGPROMPTLISTOPTION(PREFINDEX,PREFUSER(PREFINDEX))
+   END
+   DIALOGPROMPTLISTEND
+   DIALOGDISPLAY
+   SELECTEDPREF=ENTERNUMBER("Choose Banno User",1)
+   DIALOGCLOSE
+  END
+END [PROCEDURE]
+
+PROCEDURE GETPREFERENCERECS
+[* Collects needed data from each HB preference on current
+** account.
+*]
+ PREFCOUNT=0
+ FOR EACH PREFERENCE WITH (PREFERENCE:HBUSERNAME<>"" AND
+                           PREFERENCE:HBMODE=STDHOMEBANKING)
+  DO
+   PREFCOUNT=PREFCOUNT+1
+   PREFLOC(PREFCOUNT)=PREFERENCE:LOCATOR
+   PREFUSER(PREFCOUNT)=PREFERENCE:HBUSERNAME
+   PREFEMAILADDR(PREFCOUNT)=PREFERENCE:EMAILADDRESS
+   PREFPASSPROB(PREFCOUNT)=PREFERENCE:PWDRESETPROBATION
+   PREFXFERANYLOC(PREFCOUNT)=0
+   FOR EACH PREFERENCE ACCESS WITH (PREFERENCE ACCESS:ACCESSTYPE=3)
+    DO
+     PREFXFERANYLOC(PREFCOUNT)=PREFERENCE ACCESS:LOCATOR
+    END
+   UNTIL PREFXFERANYLOC(PREFCOUNT)<>0
+  END
+ IF PREFCOUNT>PREFMAX THEN
+  DO
+   POPUPMESSAGE(2,"Too many users, Update Manually")
+   TERMINATE
+  END
+END [PROCEDURE]
+
+PROCEDURE CHOOSEACTION
+[* Dialog that prompts user for desired action.
+*]
+ DIALOGSTART("Banno Enrollment",1500%,0)
+ DIALOGPROMPTLISTSTART("Choose Action:",ACTIONCHGPASS)
+ IF PREFCOUNT>=1 THEN
+  DIALOGPROMPTLISTOPTION(ACTIONCHGPASS,"Change Password")
+ IF PREFCOUNT=0 OR ALLOWMULT=TRUE THEN
+  DIALOGPROMPTLISTOPTION(ACTIONCREATEUSR,"Create User")
+ IF PREFCOUNT>=1 THEN
+  DIALOGPROMPTLISTOPTION(ACTIONEDITUSER,"Edit User")
+ IF ALLOWDELETE=TRUE AND PREFCOUNT>=1 THEN
+  DIALOGPROMPTLISTOPTION(ACTIONDELUSER,"Delete User")
+ IF ALLOWXACCT=TRUE AND PREFCOUNT>=1 THEN
+  DO
+   DIALOGPROMPTLISTOPTION(ACTIONADDXACCT,"Add Cross Account")
+   DIALOGPROMPTLISTOPTION(ACTIONEDITXACCT,"Edit Cross Account")
+   DIALOGPROMPTLISTOPTION(ACTIONDELXACCT,"Delete Cross Account")
+  END
+ DIALOGPROMPTLISTOPTION(ACTIONEXITPGM,"Exit Banno Enrollment")
+ DIALOGPROMPTLISTEND
+ DIALOGDISPLAY
+ ACTIONREQUESTED=ENTERNUMBER("Choose Action:",ACTIONCHGPASS)
+ DIALOGCLOSE
+END [PROCEDURE]
+
+PROCEDURE CHGPASSDIALOG
+[* Dialog that prompts user for new HB password and password
+** probation preference.
+*]
+ DIALOGSTART("Banno - New Password",200%,0)
+ DIALOGINTROTEXT("Changing Password for User "+PREFUSER(SELECTEDPREF))
```

Comment:
Lines 336,339: You’re placing 2 separate calls to GETDATANUMBER(GETPARAMHBPASSWORDMAXLENGTH). It might be better to place one call, storing the value into a variable to be used in place of the existing multiple calls to lower demand on system resources. (_same in some other procedures as well_)

Reply:
Done for all GETDATA statements